nat/traversal/events_tracker.go

```diff
@@ -26,29 +27,53 @@ import (
 // EventTopic the topic that traversal events are published on
 const EventTopic = "Traversal"
 
-// EventNoEvent represents an occurrence where nothing really happened
-var EventNoEvent = Event{Name: ""}
+const eventsTrackerLogPrefix = "[traversal-events-tracker] "
 
-// EventSuccess represents a successful NAT traversal
-var EventSuccess = Event{Name: "success"}
-
-// EventFailure represents a failed NAT traversal event
-var EventFailure = Event{Name: "failure"}
+const (
+	// EmptyEventType is name of event where nothing really happened
+	EmptyEventType = ""
+	// SuccessEventType is name of event for a successful NAT traversal
+	SuccessEventType = "success"
+	// FailureEventType is name of event for a failed NAT traversal
+	FailureEventType = "failure"
+)
 
 // EventsTracker is able to track NAT traversal events
 type EventsTracker struct {
-	lastEvent Event
-	eventChan chan Event
+	lastEvent     Event
+	eventChan     chan Event
+	metricsSender metricsSender
+}
+
+type metricsSender interface {
+	SendNATMappingSuccessEvent() error
+	SendNATMappingFailEvent(err error) error
+}
+
+// BuildSuccessEvent returns new event for successful NAT traversal
+func BuildSuccessEvent() Event {
+	return Event{Type: SuccessEventType}
+}
+
+// BuildFailureEvent returns new event for failed NAT traversal
+func BuildFailureEvent(err error) Event {
+	return Event{Type: FailureEventType, Error: err}
 }
 
 // NewEventsTracker returns a new instance of event tracker
-func NewEventsTracker() *EventsTracker {
-	return &EventsTracker{eventChan: make(chan Event, 1)}
+func NewEventsTracker(metricsSender metricsSender) *EventsTracker {
+	return &EventsTracker{eventChan: make(chan Event, 1), metricsSender: metricsSender}
 }
 
 // ConsumeNATEvent consumes a NAT event
 func (et *EventsTracker) ConsumeNATEvent(event Event) {
-	log.Info("got NAT event: ", event)
+	log.Info(eventsTrackerLogPrefix, "got NAT event: ", event)
+
+	err := et.sendNATEvent(event)
+	if err != nil {
+		log.Warn(eventsTrackerLogPrefix, "sending event failed", err)
```

Comment:
We need to have an explicit delimiter between arguments. 
```suggestion
		log.Warn(eventsTrackerLogPrefix, "sending event failed: ", err)
```